packages/api-security/src/index.ts

```diff
@@ -42,7 +42,9 @@ export const createSecurityContext = ({ storageOperations }: SecurityConfig) =>
                 const tenant = context.tenancy.getCurrentTenant();
                 return tenant ? tenant.id : undefined;
             },
-            storageOperations
+            storageOperations,
+            listGroupsFromPluginsCallback: () => context.plugins.byType("security-role"),
+            listTeamsFromPluginsCallback: () => context.plugins.byType("security-team")
```

Comment:
Several thoughts here:
1) Let's avoid using the `Callback` suffix, as it's really meaningless. Typescript is here to tell you that it's a function.  It can simply be `listGroupsFromPlugins`.
2) Assuming we go with `listGroupsFromPlugins` in the first bullet point, let's elaborate on that: if we talk about the _purpose_ of that parameter, it is 1) an "optional dependency", and 2) it is a "provider" because it provides you with a list of groups. 
3) Moving further, we want to code against _abstractions_. But `listGroupsFromPluginsCallback` sounds extremely specific: it tells you that this provider HAS TO load groups _from plugins_. What if I don't? How can you enforce this? You can't, all you care about is that you get back a list of Groups/Teams.

So, taking all these points into account, our code can be improved by doing a few simple changes:
1) make it more abstract, by renaming `listGroupsFromPluginsCallback` to `groupsProvider` (very abstract, can be implemented in many ways).
2) Make sure providers are async, so `groupsProvider: () => Promise<Group[]>`.
3) Same logic applies for teams, so `teamsProvider: () => Promise<Team[]>`.

Reply:
Done.